uni/lib/controller/fetchers/course_units_fetcher/all_course_units_fetcher.dart

```diff
@@ -10,18 +10,22 @@ class AllCourseUnitsFetcher {
     Session session, {
     List<CourseUnit>? currentCourseUnits,
   }) async {
-    final allCourseUnits = <CourseUnit>[];
-
-    for (final course in courses) {
-      final courseUnits = await _getAllCourseUnitsAndCourseAveragesFromCourse(
-        course,
-        session,
-        currentCourseUnits: currentCourseUnits,
-      );
-      allCourseUnits.addAll(courseUnits.where((c) => c.enrollmentIsValid()));
-    }
+    final courseCourseUnits = await Future.wait(
+      courses
+          .map(
+            (course) => _getAllCourseUnitsAndCourseAveragesFromCourse(
+              course,
+              session,
+              currentCourseUnits: currentCourseUnits,
+            ),
+          )
+          .toList(),
+    );
```

Comment:
Since the number of courses here is unbounded, we should find a way to only have 2 or 3 requests active at a time and then go through the list as futures are completed. Having too many active pending requests may slow down the process and/or cause some to fail

Reply:
Any suggestion to do so? Other than taking chunks of 3 e.g.?

Reply:
I'm searching for solutions on the web. Unfortunately, it doesn't seem the standard library offers a good way to do this :confused:

Reply:
https://pub.dev/documentation/pool/latest/pool/Pool/Pool.html

The `pool` package appears to be a good fit for this. Do we install it or do you recon it's not worth it?

Reply:
That seems interesting. What do you think of doing this globally at the `NetworkRouter` level instead?

Reply:
That would be even better. However, in that case, I think it would be better if we had something where we could configure the priority of the request.

Reply:
I'm not sure whether that would require a lock or not (for pushing to a global sorted queue). Anyway, should be investigated on its own separate issue

Reply:
I think implementing it at the `NetworkRouter` should be alright. Do create an issue about the priorities if you think it's an idea worth implementing in the future.